tests/unit-core/test-container.c

```diff
@@ -0,0 +1,86 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript.h"
+#include "test-common.h"
+
+int
+main (void)
+{
+  jerry_init (JERRY_INIT_EMPTY);
+
+  if (!jerry_is_feature_enabled (JERRY_FEATURE_MAP)
+      || !jerry_is_feature_enabled (JERRY_FEATURE_SET)
+      || !jerry_is_feature_enabled (JERRY_FEATURE_WEAKMAP)
+      || !jerry_is_feature_enabled (JERRY_FEATURE_WEAKSET))
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Containers are disabled!\n");
+    jerry_cleanup ();
+    return 0;
+  }
+
+  jerry_value_t instance_check;
+  jerry_value_t global = jerry_get_global_object ();
+  jerry_value_t map_str = jerry_create_string ((jerry_char_t *) "Map");
+  jerry_value_t set_str = jerry_create_string ((jerry_char_t *) "Set");
+  jerry_value_t weakmap_str = jerry_create_string ((jerry_char_t *) "WeakMap");
+  jerry_value_t weakset_str = jerry_create_string ((jerry_char_t *) "WeakSet");
+  jerry_value_t global_map = jerry_get_property (global, map_str);
+  jerry_value_t global_set = jerry_get_property (global, set_str);
+  jerry_value_t global_weakmap = jerry_get_property (global, weakmap_str);
+  jerry_value_t global_weakset = jerry_get_property (global, weakset_str);
+
+  jerry_release_value (global);
+
+  jerry_release_value (map_str);
+  jerry_release_value (set_str);
+  jerry_release_value (weakmap_str);
+  jerry_release_value (weakset_str);
+
+  jerry_value_t empty_map = jerry_create_container (NULL, 0, JERRY_CONTAINER_TYPE_MAP);
```

Comment:
All of these are empty containers, could we have tests for containers with contents? If so it would be good to have test functions for them. Part of the testing will be probably in JS as some of the container operations are not in the API.

Reply:
The API function part is covered by these tests, calling them with initial elements would cover the same part as our JS tests do, since the builtin routines (e.g. `ecma_builtin_map_dispatch_construct`) are calling the same functions as these API functions do (e.g. `ecma_op_container_create`).